features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/App.js

```diff
@@ -68,11 +76,22 @@ class Protected extends Component {
             );
         }
         let params = qs.stringify({referrer: this.props.location.pathname});
-        return (
-            <Redirect to={{pathname: '/login', search: params}}/>
-        );
+        if (this.state.authConfigs) {
+            if (this.state.authConfigs.is_sso_enabled) {
+                var authorizationEndpoint = this.state.authConfigs.authorizationEndpoint;
+                var client_id = this.state.authConfigs.client_id;
+                var callback_URL = this.state.authConfigs.callback_url;
+                var scopes = this.state.authConfigs.scopes;
+                window.location = authorizationEndpoint+"?response_type=code&client_id="+client_id+"&redirect_uri="+callback_URL+"&scope="+scopes;
+            } else {
+                return (
+                            <Redirect to={{pathname: '/login', search: params}}/>
+                        );
+            }
+        } else {
+            return null;
+        }
```

Comment:
returning null will stop the component from rendering AFAIK. Should display an error to the user IMHO.